### PR TITLE
Trainee Subject Show

### PR DESCRIPTION
**Pull request by shakil-fg**

### CODE REVIEW

**Comment by mdmahmudurrahman** [comment]
Ok

File: app/controllers/trainee/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    @tasks.each do |task|
+      @checked << cnt if @user.tasks.include? task
+      cnt+=1
+    end
+  end
+
+  def load_status
+    load_subject
+    @status=[]
+    course_subject=@user.course_subjects.find_by(subject_id: @subject.id)
```
Comment (MuhammadTamzid) [inline app/controllers/trainee/subjects_controller.rb]: remove ()
find_by subject_id: @subject.id

File: app/controllers/subjects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+class SubjectsController < ApplicationController
```
Comment (dieunb) [inline app/controllers/subjects_controller.rb]: `UserSubjectsController` pls

File: app/controllers/user_subjects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
+class UserSubjectsController < ApplicationController
+  before_action :load_checkbox_contents
+  before_action :load_status
+
+  def index
+    @user_subjects = @user.user_subjects
+  end
+
+  def show
+    if params["user_subject"]
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: don't push some action hit database on this action, pls move into update action.

Use callback to update task of user_subject

File: app/controllers/user_subjects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    @user_subjects = @user.user_subjects
+  end
+
+  def show
+  end
+
+  def update
+    begin
+      @user.user_tasks.each do |user_task|
+        user_task.destroy if user_task.task.subject==@user_subject.subject
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: why destroy?

File: app/controllers/user_subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      end
+      params["user_subject"]["task_ids"].each do |id|
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: user callback pls

File: app/controllers/user_subjects_controller.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      flash[:success] = t :task_update_success
+    rescue
+      flash[:danger] = t :task_update_failure
+    end
+    redirect_to @user_subject
+  end
+
+  private
+  def load_user
+    @user = User.first # Incompatibility 1: Please see SpecificationNotes.md
```
Comment (noor-framgia) [inline app/controllers/user_subjects_controller.rb]: maybe it is solved now

File: app/controllers/user_subjects_controller.rb [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  def index
+    @user_subjects = @user.user_subjects
+  end
+
+  def show
+  end
+
+  def update
+    begin
+      tasks=[]
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: This line code, so it finish tasks, so update user_subject with task via nested attributes 
```
+      tasks=[]
+      user_subject_params.each do |id|
+        tasks<<Task.find(id)
+      end
+      @user.tasks=tasks
```

Comment (shakil-fg) [inline app/controllers/user_subjects_controller.rb]: I am not seeing an obvious way to implement nested attribute here. Cause till now we haven't implemented nested attributes on relationship models. Please guide me to implement it.

This is the link to our DB https://www.chatwork.com/gateway.php?cmd=download_file&bin=1&file_id=60790621.

Comment (shakil-fg) [inline app/controllers/user_subjects_controller.rb]: Modified the submit button to conditionally appear when user_subject is not finished

**Pull request merged**